### ISSUE TITLE
Support custom moduleDirectories

### ISSUE THREAD

**Issue by taion** [postcss/postcss-import#195]
In 7.x, the specified `path`(s) functioned as additional module directories.

I used this to follow the "huge apps" example in using nested `shared/` directories, that were resolved as module directories.

Was there a specific reason for its removal? It's not a feature I've made heavy use of, but it seems like an odd removal. Is the idea that this feature is superseded by the custom `resolve` argument?

**Comment by MoOx** [comment]
> Is the idea that this feature is superseded by the custom resolve argument?

I think so. I have to admit I don't really use this package except for ``node_modules`` and ``web_modules``.

@TrySound handled the 8.x release (huge refactoring). It seems there is some major issues tho like #168 so I may consider simply reverting to 7.x implementation (which is simpler and don't have this kind of bugs) since nobody is currently maintaining this package at the moment (and I found the current codebase more complicated, and API too low level).

**Comment by taion** [comment]
FWIW, this ended up being easy to work around for me, since apparently I was only using the moduleDirectories thing for a single directory.

**Comment by RyanZim** [comment]
> it seems like an odd removal

Agreed.

I would be in favor of adding this back; however, I'm not sure of the best route to proceed. Should custom moduleDirectories replace or be appended to the default `[ 'node_modules', 'web_modules' ]`?

@MoOx @TrySound @ai Thoughts?

**Comment by MoOx** [comment]
I would go for appending them. I guess nobody will want to remove those.

**Comment by RyanZim** [comment]
> I guess nobody will want to remove those.

Yeah, if they do, they can use `resolve`.

**Comment by neezer** [comment]
Voicing my support for this. I'm using the [application structure](https://gist.github.com/ryanflorence/daafb1e3cb8ad740b346) outlined by @ryanflorence, which relies on being able to customize `moduleDirectories`. Unlike @taion, I am actually using this feature pretty heavily. 😉 

In the meantime, I've just forked this repo and [added the directory manually](https://github.com/postcss/postcss-import/compare/master...neezer:master), which unblocks me until any official update.

**Comment by RyanZim** [comment]
@neezer Thanks for chiming in. I want to get v9 shipped today or tomorrow; then this will be PR welcome for the next release.

**Comment by RyanZim** [comment]
@neezer @taion v9 is shipped, PR welcome.

**Comment by taion** [comment]
We ended up not using this pattern – the reason ended up being that we use CSS modules, for which modules directories support already works fine, and only used `@import` for variables. Those variables we wanted available in both CSS and JS anyway, so we ended up just declaring them all at top-level in JS for shared use across everything.

If you're using CSS modules, I'd recommend using https://github.com/css-modules/postcss-modules-values anyway.
